File: PRSWeb/Models/PurchaseRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;

namespace PRSWeb.Models {
	public class PurchaseRequest {
		public int Id { get; set; }

		[StringLength(80)]
		[Required]
		public string Description { get; set; }

		[StringLength(160)]
		[Required]
		public string Justification { get; set; }

		[Required]
		public DateTime DateNeeded { get; set; }

		[StringLength(30)]
		[Required]
		public string DeliveryMode { get; set; }

		[StringLength(30)]
		[Required]
		public string Status { get; set; }

		[Required]
		public double Total { get; set; }

		[Required]
		public DateTime SubmittedDate { get; set; }

		public int UserId { get; set; }
		public virtual User User { get; set; }
	}
}

File: ng/src/app/models/PurchaseRequest.ts
export class PurchaseRequest {
	
	Id: number;
	Description: string;
	Justification: string;
	RejectionReason: string;
	DeliveryMode: string;
	DateNeeded: Date;
	SubmittedDate: Date;
	Status: string;
	Total: number;
	UserId: number;

	constructor(
		Id: number,
		Description: string,
		Justification: string,
		RejectionReason: string,
		DeliveryMode: string,
		DateNeeded: Date,
		SubmittedDate: Date,
		Status: string,
		Total: number,
		UserId: number
	) {
		this.Id = Id;
		this.Description = Description;
		this.Justification = Justification;
		this.RejectionReason = RejectionReason;
		this.DeliveryMode = DeliveryMode;
		this.DateNeeded = DateNeeded;
		this.SubmittedDate = SubmittedDate;
		this.Status = Status;
		this.Total = Total;
		this.UserId = UserId;
	}
}